main.go

```diff
@@ -186,87 +186,71 @@ func getSecretGenerators(c *controller) []SecretGenerator {
 	return secretGenerators
 }
 
-func (c *controller) process() error {
-	secretGenerators := getSecretGenerators(c)
-
-	for _, secretGenerator := range secretGenerators {
+func (c *controller) processNamespace(namespace *v1.Namespace, secret *v1.Secret) error {
+	// Check if the secret exists for the namespace
+	_, err := c.k8sutil.GetSecret(namespace.GetName(), secret.Name)
 
-		fmt.Printf("------------------ [%s] ----------------------\n", secretGenerator.SecretName)
-
-		newToken, err := secretGenerator.TokenGenFxn()
+	if err != nil {
+		// Secret not found, create
+		err := c.k8sutil.CreateSecret(namespace.GetName(), secret)
 		if err != nil {
-			fmt.Printf("Error getting secret for provider %s. Skipping secret provider! [Err: %s]", secretGenerator.SecretName, err)
-			continue
+			return fmt.Errorf("Could not create Secret! %v", err)
 		}
-		newSecret := generateSecretObj(newToken.AccessToken, newToken.Endpoint, secretGenerator.IsJSONCfg, secretGenerator.SecretName)
-
-		namespaces, err := c.k8sutil.GetNamespaces()
-
+	} else {
+		// Existing secret needs updated
+		err := c.k8sutil.UpdateSecret(namespace.GetName(), secret)
 		if err != nil {
-			fmt.Println("-------> ERROR getting namespaces! Skipping secret provider!", err)
-			continue
+			return fmt.Errorf("Could not update Secret! %v", err)
 		}
+	}
+
+	// Check if ServiceAccount exists
+	serviceAccount, err := c.k8sutil.GetServiceAccount(namespace.GetName(), "default")
+	if err != nil {
+		return fmt.Errorf("Could not get ServiceAccounts! %v", err)
```

Comment:
I changed the logic from before where now I think its best to try and process each cloud (e.g. ECR & GCR right now). We need to make sure if one cloud fails, it doesn't bomb out the other. This is where I've had a bunch of `continue`s in there to keep the process running.